vshard/replicaset.lua

```diff
@@ -1204,6 +1204,9 @@ local replica_mt = {
         safe_uri = function(replica)
             local uri = luri.parse(replica.uri)
             uri.password = nil
+            if uri.ipv6 then
+                uri.host = '[' .. uri.host .. ']'
+            end
             return luri.format(uri)
```

Comment:
Nit: in the future, when https://github.com/tarantool/tarantool/issues/9556 is fixed, we'll have to make this brackets adding depend on Tarantool's version, as it's actually a bug in `uri.format` function.

So, I'd rather move this check (for brackets) alongside with `luri.format` to `utils.lua` and name it smth like `uri_format`. This way we'll have to make minimal changes in order to introduce version specific `uri_format` function. Up to you and @Gerold103, I'm ok with the current implementation too.

Moreover, we should create a ticket for introducing version dependent `uri_format` function, when this PR is merged.

Reply:
I'd like to see these changes merged as soon as possible to get them into the next release.